src/frontend/src/handler/create_function.rs

```diff
@@ -173,16 +171,56 @@ pub async fn handle_create_function(
         }
         "javascript" => {
             identifier = function_name.to_string();
-            body = Some(match params.as_ {
-                Some(FunctionDefinition::SingleQuotedDef(s)) => s,
-                Some(FunctionDefinition::DoubleDollarDef(s)) => s,
-                _ => {
-                    return Err(ErrorCode::InvalidParameterValue(
-                        "AS must be specified".to_string(),
-                    )
-                    .into())
-                }
-            });
+            body = Some(
+                params
+                    .as_
+                    .ok_or_else(|| ErrorCode::InvalidParameterValue("AS must be specified".into()))?
+                    .into_string(),
+            );
+        }
+        "rust" => {
+            identifier = wasm_identifier(
+                &function_name,
+                &arg_types,
+                &return_type,
+                matches!(kind, Kind::Table(_)),
+            );
+            if params.using.is_some() {
+                return Err(ErrorCode::InvalidParameterValue(
+                    "USING is not supported for rust function".to_string(),
+                )
+                .into());
+            }
+            let function_body = params
+                .as_
+                .ok_or_else(|| ErrorCode::InvalidParameterValue("AS must be specified".into()))?
+                .into_string();
+            let script = format!("#[arrow_udf::function(\"{identifier}\")]\n{function_body}");
+            body = Some(function_body.clone());
+
+            let wasm_binary =
+                tokio::task::spawn_blocking(move || arrow_udf_wasm::build("", &script))
```

Comment:
IIUC, this means that we don't support specifying external crate dependencies now. This sounds good as the `build.rs` could run arbitrary code and is quite unsafe in a managed environment.

I'm considering whether it's possible to support a whitelist of 3rd-party crates in the future, just like how Rust's online playground does: https://arc.net/l/quote/qtopwevk

Reply:
Or if `wasm` can serve as a with-host-tools toolchain in the distant future... 🤣

https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-2-with-host-tools

Reply:
> I'm considering whether it's possible to support a whitelist of 3rd-party crates in the future, just like how Rust's online playground does: https://arc.net/l/quote/qtopwevk

Good idea! We can follow their way: pre-download all the crates that Rust playground supports and add `--offline` to `cargo build`.